components/mediation/data-publishers/org.wso2.micro.integrator.prometheus.publisher/src/main/java/org/wso2/micro/integrator/prometheus/publisher/service/MetricResource.java

```diff
@@ -85,4 +99,18 @@ public boolean invoke(MessageContext synCtx) {
         return true;
     }
 
+    private Set<String> parseQuery(String query) throws IOException {
+
+        Set<String> names = new HashSet<>();
+        if (query != null) {
+            String[] pairs = query.split("&");
+            for (String pair : pairs) {
+                int idx = pair.indexOf("=");
+                if (idx != -1 && URLDecoder.decode(pair.substring(0, idx), "UTF-8").equals("name[]")) {
+                    names.add(URLDecoder.decode(pair.substring(idx + 1), "UTF-8"));
```

Comment:
I think we need to break the loop here

Reply:
A user can use multiple names to filter the metrics. So, I think breaking the loop should not be done here.